identity/test/controllers/EditProfileControllerTest.scala

```diff
@@ -88,6 +88,46 @@ class EditProfileControllerTest extends path.FreeSpec with ShouldMatchers with M
     }
   }
 
+  "Given the submitPrivacyForm method is called" - {
+    val receive3rdPartyMarketing = false
+    val receiveGnmMarketing = true
+
+    "with a valid CSRF request" - Fake{
+      val fakeRequest = FakeCSRFRequest()
+        .withFormUrlEncodedBody(
+          "receiveGnmMarketing" -> receiveGnmMarketing.toString,
+          "receive3rdPartyMarketing" -> receive3rdPartyMarketing.toString
+        )
+
+      val updatedUser = user.copy(
+        statusFields = StatusFields(
+          receiveGnmMarketing = Some(receiveGnmMarketing),
+          receive3rdPartyMarketing = Some(receive3rdPartyMarketing)
+        )
+      )
+      when(api.saveUser(Matchers.any[String], Matchers.any[UserUpdate], Matchers.any[Auth]))
+        .thenReturn(Future.successful(Right(updatedUser)))
+
+      val result = controller.submitPrivacyForm().apply(fakeRequest)
+
+      Await.result(result, 10.seconds)
```

Comment:
Even though this probably wont fail, 10 seconds is quite long to wait, could this be just 1 second?